flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/history/FsJobArchivistTest.java

```diff
@@ -30,56 +29,32 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.regex.Pattern;
+
+import static org.hamcrest.core.IsCollectionContaining.hasItem;
 
 /**
  * Tests for the {@link FsJobArchivist}.
  */
 public class FsJobArchivistTest {
 
 	@Rule
-	public TemporaryFolder tmpFolder = new TemporaryFolder();
+	public final TemporaryFolder tmpFolder = new TemporaryFolder();
 
 	@Test
 	public void testArchiveJob() throws Exception {
-		Path tmpPath = new Path(tmpFolder.getRoot().getAbsolutePath());
-
-		AccessExecutionGraph graph = ArchivedJobGenerationUtils.getTestJob();
-
-		Collection<ArchivedJson> expected = new ArrayList<>();
-		for (JsonArchivist archivist : Collections.<JsonArchivist>emptyList()) {
-			for (ArchivedJson archive : archivist.archiveJsonWithPath(graph)) {
-				expected.add(archive);
-			}
-		}
-
-		Path archivePath = FsJobArchivist.archiveJob(tmpPath, graph);
-		Collection<ArchivedJson> actual = FsJobArchivist.getArchivedJsons(archivePath);
+		final Path tmpPath = new Path(tmpFolder.getRoot().getAbsolutePath());
+		final JobID jobId = new JobID();
 
-		Assert.assertEquals(expected.size(), actual.size());
+		final Collection<ArchivedJson> toArchive = new ArrayList<>(2);
+		toArchive.add(new ArchivedJson("dir1", "hello"));
+		toArchive.add(new ArchivedJson("dir1/dir11", "world"));
 
-		Iterator<ArchivedJson> eI = expected.iterator();
-		Iterator<ArchivedJson> aI = actual.iterator();
+		final Path archive = FsJobArchivist.archiveJob(tmpPath, jobId, toArchive);
+		final Collection<ArchivedJson> restored = FsJobArchivist.getArchivedJsons(archive);
 
-		// several jsons contain a dynamic "now" field that depends on the time of creation, so we can't easily compare
-		// the json and only check the path
-		// /jobs/:jobid
-		// /jobs/:jobid/vertices
-		// /jobs/:jobid/vertices/:vertexid
-		// /jobs/:jobid/vertices/:vertexid/subtasktimes
-		// /jobs/:jobid/vertices/:vertexid/taskmanagers
-		Pattern jobDetailsPattern = Pattern.compile("/jobs/[a-fA-F0-9]{32}(/vertices(/[a-fA-F0-9]{32}(/(subtasktimes|taskmanagers))?)?)?");
-		while (eI.hasNext() && aI.hasNext()) {
-			// technically there isn't guarantee that the order is identical, but as it stands this is the case
-			ArchivedJson exp = eI.next();
-			ArchivedJson act = aI.next();
-			if (jobDetailsPattern.matcher(exp.getPath()).matches()) {
-				Assert.assertEquals(exp.getPath(), act.getPath());
-			} else {
-				Assert.assertEquals(exp, act);
-			}
+		Assert.assertEquals(toArchive.size(), restored.size());
+		for (final ArchivedJson json : toArchive) {
```

Comment:
Looks like it could be implemented as

```
import static org.hamcrest.Matchers.containsInAnyOrder;
import static org.junit.Assert.assertThat;

...

assertThat(restored, containsInAnyOrder(toArchive.toArray()));
```

Checking the size of the collection wouldn't be needed then:
> Creates an order agnostic matcher for Iterables that matches when a single pass over the examined Iterable yields a series of items, each logically equal to one item anywhere in the specified items. For a positive match, the examined iterable must be of the same length as the number of specified items.
N.B. each of the specified items will only be used once during a given examination, so be careful when specifying items that may be equal to more than one entry in an examined iterable.

Reply:
one could probably do that. I kind of do not wanna do this collection to array conversion because it is just.so.stupid; internally it is again converted into a collection -.-

Reply:
Just pick what you deem more readable. Another option:

```
final ArchivedJson archivedJson1 = new ArchivedJson("dir1", "hello");
final ArchivedJson archivedJson2 = new ArchivedJson("dir1/dir11", "world");

final Path archive = FsJobArchivist.archiveJob(tmpPath, jobId, Arrays.asList(archivedJson1, archivedJson2));
final Collection<ArchivedJson> restored = FsJobArchivist.getArchivedJsons(archive);

assertThat(restored, containsInAnyOrder(archivedJson1, archivedJson2));
```